### ISSUE TITLE
server v3 crash

### ISSUE THREAD

**Issue by Izumiko** [RikkaApps/Shizuku#30]
Device: HTC U11 (Japan)
Android 8.0 without root

log:
```
D:\Tools\platform-tools>adb shell sh /sdcard/Android/data/moe.shizuku.privileged.api/files/start.sh
info: start.sh begin
chcon: '/data/local/tmp/shizuku_starter' to u:object_r:shell_data_file:s0: Permission denied
info: path is /storage/emulated/0/Android/data/moe.shizuku.privileged.api/files/server-v3.dex
info: path-legacy is /storage/emulated/0/Android/data/moe.shizuku.privileged.api/files/server-legacy-v32-api26.dex
info: path is /data/local/tmp/shizuku/server-v3.dex
info: path-legacy is /data/local/tmp/shizuku/server-legacy-v32-api26.dex
info: starter begin
info: killing old process...
info: killed 20131 (shizuku_server_legacy)
info: starting server v3...
info: process forked, pid=20933
info: checking shizuku_server start...
warn: timeout but can't get pid of shizuku_server.
--- crash start ---
[command] logcat -b crash -t '03-27 22:19:52.000' -d
03-27 22:19:52.950 20934 20934 E AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: main
03-27 22:19:52.950 20934 20934 E AndroidRuntime: java.lang.SecurityException: Permission Denial: registerUidObserver from pid=20934, uid=2000 requires android.permission.PACKAGE_USAGE_STATS
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at android.os.Parcel.readException(Parcel.java:1960)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at android.os.Parcel.readException(Parcel.java:1906)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at android.app.IActivityManager$Stub$Proxy.registerUidObserver(IActivityManager.java:9128)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at moe.shizuku.server.api.Api.registerUidObserver(Api.java:110)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at moe.shizuku.server.BinderSender.register(BinderSender.java:119)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at moe.shizuku.server.ShizukuService.<init>(ShizukuService.java:52)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at moe.shizuku.server.Starter.main(Starter.java:109)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-27 22:19:52.950 20934 20934 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:284)
03-27 22:19:52.951 20934 20934 E AndroidRuntime: Error reporting crash
03-27 22:19:52.951 20934 20934 E AndroidRuntime: java.lang.RuntimeException: Bad file descriptor
03-27 22:19:52.951 20934 20934 E AndroidRuntime:        at android.os.BinderProxy.transactNative(Native Method)
03-27 22:19:52.951 20934 20934 E AndroidRuntime:        at android.os.BinderProxy.transact(Binder.java:756)
03-27 22:19:52.951 20934 20934 E AndroidRuntime:        at android.app.IActivityManager$Stub$Proxy.handleApplicationCrash(IActivityManager.java:4345)
03-27 22:19:52.951 20934 20934 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$KillApplicationHandler.uncaughtException(RuntimeInit.java:115)
03-27 22:19:52.951 20934 20934 E AndroidRuntime:        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
03-27 22:19:52.951 20934 20934 E AndroidRuntime:        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1063)
03-27 22:19:52.951 20934 20934 E AndroidRuntime:        at java.lang.Thread.dispatchUncaughtException(Thread.java:1953)
--- crash end ---
--- shizuku start ---
[command] logcat -b main -t '03-27 22:19:52.000' -d -s ShizukuServer ShizukuServerV3 ShizukuManager
03-27 22:19:52.940 20934 20934 I ShizukuServerV3: server v3
03-27 22:19:52.940 20934 20934 I ShizukuServerV3: using token from arg
--- shizuku end ---
info: shizuku_starter exit with non-zero value 7
```

**Comment by RikkaW** [comment]
Should fixed in v3.2.1

**Comment by zary38231-spec** [comment]
120 Fp
